src/app/my-account/my-account.component.ts

```diff
@@ -56,25 +53,18 @@ export class MyAccountComponent implements OnInit {
       this.messages.message = 'New Passwords Do Not Match!';
     } else if (this.passFields.oldPassField === this.passFields.newPassField) {
       this.messages.message = 'Choose A New Password!';
-    } else if (!this.oldPassMatches()) {
-      this.passColors.oldPass = this.colors.red;
-      this.messages.message = 'Old Password Is Invalid!';
     } else {
-      // TODO: REMOVE the following line when we connect to DB
-      this.pword = this.passFields.newPassField;
-      this.messages.message = 'Password Changed';
-      warn = false;
-      this.passColors.oldPass = this.colors.green;
+      this.apiUsersCrudService.changePassword(this.passFields.oldPassField, this.passFields.newPassField).subscribe(
+        (_) => {
+          this.messages.message = 'Password Changed';
+          warn = false;
+          this.passColors.oldPass = this.colors.green;
+        });
```

Comment:
If the old password doesn't match there should be a message and password field color change, similar to how validating the rest of it works.
```suggestion
        }, () => {
          this.passColors.oldPass = this.colors.red;
          this.messages.message = 'Old Password Is Invalid!';
        });
```